examples/text_labels_and_annotations/mathtext_fontset_example.py

```diff
@@ -0,0 +1,27 @@
+"""
+========================
+Mathtext Fontset Example
+========================
+
+A simple example of using different font families for a math text.
+"""
+
+import matplotlib.pyplot as plt
+
+plt.figure(figsize=(12, 10))
+
+#A simple plot for the background
+plt.plot(range(11), color="0.9", label=r'Fontset Example')
+
+#The texts to be shown
+plt.text(0.4, 5, r"$\mathit{This\ text\ should\ have\ one\ font} $",
+         size=24, fontset='dejavusans')
+
+#We can use either fontset or mathtext_fontset as a parameter for the font
```

Comment:
I can't tell why I'd use `fontset` or `mathtext_fontset` from this example since the both seem to be doing the same thing to me.

Reply:
like I said in my comment above, the fontProperties takes the value of rcParams at the moment it is created, so if you do something like :
```
rcParams['mathtext.fontset] = 'dejavusans'
plt.text(0.4, 5, r"$\mathit{some\ text} $" )
rcParams['mathtext.fontset] = 'dejavuserif'
plt.text(0.4, 5, r"$\mathit{some\ other\ text} $" )
plt.show()
```
You will get the same result as using the `fontset` parameter. This was not intentional but I believe its a good behavior. If you tried to run the same code before my implementations you would get texts in just one font (the latest set), this was the change asked in issue #17906 .